packages/provider/src/transports/base-provider-transport.ts

```diff
@@ -224,12 +224,15 @@ export abstract class BaseProviderTransport implements ProviderTransport {
 
     // ACCOUNTS_CHANGED -- when a user logs in or out
     if (message.type === EventType.ACCOUNTS_CHANGED) {
-      this.accountsChangedPayload = undefined
+      this.accountsChangedPayload = { accounts: [] }
       if (message.data && message.data.length > 0) {
-        this.accountsChangedPayload = ethers.utils.getAddress(message.data[0])
-        this.events.emit('accountsChanged', [this.accountsChangedPayload])
+        this.accountsChangedPayload = {
+          accounts: [ethers.utils.getAddress(message.data[0])],
+          origin: message.origin
+        }
+        this.events.emit('accountsChanged', this.accountsChangedPayload.accounts, this.accountsChangedPayload.origin)
```

Comment:
I wonder if we should just emit `this.accountsChangedPayload` ? I suppose this could be a breaking change which is the problem, is it more correct according to EIP1193 and what other consumers are expecting?

Reply:
yeah this is mainly about `TypedEventEmitter.emit` method and needing to append things so that i will not break current usages. The consumers are dapps connected and the wallet webapp.